curp/src/client.rs

```diff
@@ -262,18 +289,30 @@ where
             let resp = match resp_result {
                 Ok(resp) => resp.into_inner(),
                 Err(e) => {
+                    let meta = e.metadata();
+                    if let Some(label) = meta.get("error-label") {
+                        if "shutting-down"
+                            == label.to_str().unwrap_or_else(|err| {
+                                unreachable!(
+                                    "error-label should be always able to convert to str: {err:?}"
+                                )
+                            })
```

Comment:
We could make a function to extract meta str as it is used many times.

Reply:
Done